jgrapht-core/src/main/java/org/jgrapht/traverse/LexBreadthFirstIterator.java

```diff
@@ -132,7 +148,70 @@ public LexBreadthFirstIterator(Graph<V, E> graph,
         GraphTests.requireUndirected(graph);
 
         Set<V> copyOfSet = new HashSet<>(graph.vertexSet());
+        labelVertices(graph.vertexSet());
         bucketList = new BucketList(copyOfSet, priorityA, priorityB, neighborIndexA, neighborIndexB, ASets, BSets);
+        
+        // Invert priorities
+        HashMap<Integer, V> inversePriorityA = new HashMap<Integer, V>();
+        HashMap<Integer, V> inversePriorityB = new HashMap<Integer, V>();
+        
+        for(V vertex : graph.vertexSet()) {
+            inversePriorityA.put(priorityA.get(vertex), vertex);
+            inversePriorityB.put(priorityB.get(vertex), vertex);
+        }
+        
+        // Precompute sorted neighborhoods
+        this.sortedNeighbors = computeSortedNeighborhoods(graph, inversePriorityA);
+        this.sortedNeighborsB = computeSortedNeighborhoods(graph, inversePriorityB);
+    }
+    
+    /**
+     * Helper function to compute the neighborhoods sorted by the vertex priority for efficiency.
+     * 
+     * @param graph The graph
```

Comment:
The only graph. The one and only! (which one?)

Reply:
:), done.